### PR TITLE
switch to ansible 2.8

### PR DESCRIPTION
**Pull request by guits**
- remove private attribute with import_role.
- update documentation.
- update rpm spec requirement.
- fix MagicMock python import in unit tests.

Closes: #3765 

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

**Comment by guits** [comment]
jenkins test dev-centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test dev-centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test dev-centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test dev-centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test dev-centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test dev-centos-container-all_daemons

**Comment by dsavineau** [comment]
For the test_ceph_volume issue, this is related to the move from `ansible.compat.tests` to `units.compat`

We also need to remove:
  - `static: False` from roles/ceph-osd/tasks/main.yml
  - `private: fals` from site-container.yml.sample (with a typo)

**Comment by guits** [comment]
thanks maggle!

**Comment by dsavineau** [comment]
jenkins test dev-centos-non_container-all_daemons

**Comment by dsavineau** [comment]
jenkins test dev-centos-container-all_daemons

**Comment by guits** [comment]
jenkins test dev-centos-non_container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test dev-centos-container-lvm_osds

**Comment by dsavineau** [comment]
Seems like we need to update testinfra...

```console
/tmp/venv.vL09Zj4Tzh/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/testinfra/utils/ansible_runner.py:153: in __init__
    self.cli.options = self.cli.base_parser(
E   AttributeError: 'PlaybookCLI' object has no attribute 'base_parser'
```

**Comment by dsavineau** [comment]
jenkins test dev-centos-container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test dev-centos-container-lvm_osds

**Comment by dsavineau** [comment]
@guits I've bump the PR for 2.8.0rc1

**Comment by dsavineau** [comment]
@guits I've bump the PR for 2.8.0rc2

**Comment by dsavineau** [comment]
jenkins test dev-centos-non_container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test dev-centos-container-lvm_osds

File: roles/ceph-validate/tasks/check_system.yml [hunk 1]
```diff
@@ -55,15 +55,15 @@
...
   when: ansible_version.major|int < 2
 
 - name: fail on unsupported ansible version
   fail:
-    msg: "Ansible version must be 2.7 or 2.8!"
+    msg: "Ansible version must be 2.8!"
   when:
     - ansible_version.major|int == 2
-    - ansible_version.minor|int not in [7, 8]
+    - ansible_version.minor|int != 8
```
Comment (guits) [inline roles/ceph-validate/tasks/check_system.yml]: not sure whether we should remove 2.7 from ceph-validate at the moment

Comment (dsavineau) [reply roles/ceph-validate/tasks/check_system.yml]: @fultonj any thoughts on this for RDO ?

**Comment by guits** [comment]
jenkins test dev-centos-container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test dev-centos-container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test dev-centos-non_container-lvm_osds

**Comment by guits** [comment]
jenkins test dev-centos-non_container-lvm_osds

**Comment by guits** [comment]
jenkins test dev-centos-non_container-lvm_osds

**Comment by guits** [comment]
jenkins test dev-centos-non_container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test dev-centos-non_container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test dev-centos-container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test dev-centos-non_container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test dev-centos-non_container-all_daemons

**Comment by dsavineau** [comment]
jenkins test dev-centos-container-lvm_osds

**Comment by dsavineau** [comment]
jenkins test dev-centos-container-all_daemons

File: roles/ceph-validate/tasks/check_system.yml [hunk 2]
```diff
@@ -78,10 +78,14 @@
       when: ansible_distribution not in ['RedHat', 'CentOS', 'Fedora']
 
     - name: fail on unsupported distribution version for iscsi gateways
-      fail:
-        msg: "iSCSI gateways can only be deployed on Red Hat Enterprise Linux or CentOS >= 7.4"
-      when:
-        - ansible_distribution_version < '7.4'
-        - ansible_distribution in ['RedHat', 'CentOS']
+      command: 'grep -q {{ item }}=m /boot/config-{{ ansible_kernel }}'
```
Comment (guits) [inline roles/ceph-validate/tasks/check_system.yml]: this file isn't always present (atomic host):

```bash
-bash-4.2# ls /boot/
efi  grub  grub2  loader  loader.0  ostree
```

Comment (dsavineau) [reply roles/ceph-validate/tasks/check_system.yml]: Yep I saw that last friday...

**Comment by guits** [comment]
jenkins test dev-centos-container-lvm_osds

**Comment by guits** [comment]
jenkins test dev-centos-non_container-lvm_osds

**Comment by guits** [comment]
jenkins test playbook syntax

**Comment by guits** [comment]
jenkins test playbook syntax

**Comment by guits** [comment]
jenkins test playbook syntax

**Comment by guits** [comment]
jenkins test dev-centos-non_container-lvm_osds

**Comment by guits** [comment]
jenkins test dev-centos-container-all_daemons

**Comment by guits** [comment]
jenkins test pipeline

**Comment by guits** [comment]
jenkins test dev-centos-non_container-all_daemons

**Comment by dsavineau** [comment]
@guits ansible 2.8 released

**Comment by guits** [comment]
jekins test dev-centos-container-all_daemons

**Comment by guits** [comment]
failures are unrelated

**Comment by alfredodeza** [comment]
This newer version of testinfra is breaking ceph-volume tests, I believe it is related to Paramiko being unable to read the SSH config file which has host mappings for the test nodes.

See https://github.com/philpep/testinfra/commit/46457f68b4f209ddb62ca174ecaf5073a3648727

Failures look like:

```
self = <paramiko.client.SSHClient object at 0x7f81dedc4e10>, hostname = 'osd0'
port = 22

    def _families_and_addresses(self, hostname, port):
        """
        Yield pairs of address families and addresses to try for connecting.
    
        :param str hostname: the server to connect to
        :param int port: the server port to connect to
        :returns: Yields an iterable of ``(family, address)`` tuples
        """
        guess = True
        addrinfos = socket.getaddrinfo(
>           hostname, port, socket.AF_UNSPEC, socket.SOCK_STREAM
        )
E       gaierror: [Errno -2] Name or service not known
```

**Comment by alfredodeza** [comment]
Poking inside the tox.ini I figured ceph-ansible is protected with the `--ssh-config` flag, so I am passing it now and it seems to do the trick! https://github.com/ceph/ceph/pull/28294

**Pull request merged**